sqlite.go

```diff
@@ -24,21 +24,30 @@ type Dialector struct {
 	Conn       gorm.ConnPool
 }
 
+type Config struct {
+	DriverName string
+	DSN        string
+	Conn       gorm.ConnPool
+}
+
 func Open(dsn string) gorm.Dialector {
 	return &Dialector{DSN: dsn}
 }
 
+func New(config Config) gorm.Dialector {
+	return &Dialector{DSN: config.DSN, DriverName: config.DriverName, Conn: config.Conn}
+}
+
 func (dialector Dialector) Name() string {
 	return "sqlite"
 }
 
 func (dialector Dialector) Initialize(db *gorm.DB) (err error) {
-	if dialector.DriverName == "" {
-		dialector.DriverName = DriverName
-	}
 
 	if dialector.Conn != nil {
 		db.ConnPool = dialector.Conn
+	} else if dialector.DriverName != "" {
```

Comment:
Is this if branch doing anything different than the else brach below?

Reply:
No. I've removed it. I was hugely overengineering the solution to this problem!